File: app/src/main/java/com/incendiary/android/di/component/ActivityComponent.java
package com.incendiary.android.di.component;

import dagger.Subcomponent;
import com.incendiary.android.di.PerActivity;
import com.incendiary.android.di.module.ActivityModule;
import com.incendiary.android.ui.main.MainActivity;

/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity @Subcomponent(modules = ActivityModule.class) public interface ActivityComponent {
  void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/incendiary/androidboilerplate/di/component/ActivityComponent.kt
package com.incendiary.androidboilerplate.di.component

import com.incendiary.androidboilerplate.di.PerActivity
import com.incendiary.androidboilerplate.di.module.ActivityModule
import com.incendiary.androidboilerplate.features.main.MainActivity
import dagger.Subcomponent

/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity @Subcomponent(modules = arrayOf(ActivityModule::class)) interface ActivityComponent {
	fun inject(mainActivity: MainActivity)
}
